keps/78-dynamically-reclaiming-resources/README.md

```diff
@@ -0,0 +1,142 @@
+
+Hence, as seen from the above discussed Pod failure cases, we conclude that the Pods of a Job which fail during its execution, its resources are not immediately reclaimed by Kueue. Only when the Job gets marks as `Failed`, the resources of failed Pods will be reclaimed by Kueue.
+
+## API
+
+A new field `ReclaimedPodSetPods` is added to the `.status` of Workload API.
+
+```go
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+    // conditions hold the latest available observations of the Workload
+    // current state.
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    Conditions []WorkloadCondition `json:"conditions,omitempty"`
+
+    // list of count of Pods of a PodSet with resources reclaimed
+    // +optional
+    ReclaimedPodSetPods []ReclaimedPodSetPods `json:"reclaimedPodSetPods"`
+}
+
+// ReclaimedPodSetPods defines the PodSet name and count of successfully completed Pods 
+// belonging to the PodSet whose resources have been reclaimed by Kueue
+type ReclaimedPodSetPods struct{
+    Name string
+    Count int
+}
+```
+
+#### ReclaimedPodSetPods
+`.status.reclaimedPodSetPods` is a list where each element of the list denotes the count of Pods of the Workload w.r.t PodSet whose resources have been reclaimed by Kueue. The structure consists of two fields - `Name` denotes the name of PodSet and `Count` denotes the count of Pods belonging to the `Name` PodSet.
+
+## Implementation
+
+The workload reconciler will keep a watch(or `Watches()`) on the Job's `.status.succeeded` field. The Workload reconciler will calculate the difference between the Job's `.status.succeeded` field and Workload's `.status.reclaimedPodSetPods[i].count` field for every PodSet of the workload. With the former value being greater than the later, Kueue will reclaim the resources of the excess succeeded Pods.
+
+## Testing Plan
+
+Dynamically reclaiming resources enhancement has unit and integration tests. These tests
+are run regularly as a part of Kueue's prow CI/CD pipeline.
+
+### Unit Tests
+
+All the Kueue's core components must be covered by unit tests.  Here is a list of unit tests required for the modules of the feature:
+
+* [Cluster-Queue cache tests](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/cache/cache_test.go)
+* [Workload tests](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/workload/workload_test.go)
+
+### Integration tests
+* Kueue Job Controller
+  - checking the resources owned by a Job are released to the cache and clusterQueue when a Pod of the Job succeed.
+  - Integration tests for Job controller are [found here](https://github.com/kubernetes-sigs/kueue/blob/main/test/integration/controller/job/job_controller_test.go).
+
+* Workload Controller
+  - A pending Workload should be admitted when enough resources are available after release of resources by the succeeded Pods of the parallel Jobs.
```

Comment:
these tests don't have the scheduler running, so you couldn't verify this here.

Reply:
Removed